frontend/hooks/useRewardsHistory.ts

```diff
@@ -207,54 +234,50 @@ export const useRewardsHistory = () => {
   );
 
   const latestRewardStreak = useMemo<number>(() => {
+    if (isLoading || isFetching) return 0;
     if (!contractCheckpoints) return 0;
 
     // remove all histories that are not earned
-    const earnedCheckpoints = allCheckpoints.filter(
+    const earnedCheckpoints = epochSortedCheckpoints.filter(
       (checkpoint) => checkpoint.earned,
     );
 
-    // sort descending by epoch end time
-    const sorted = earnedCheckpoints.sort(
-      (a, b) => b.epochEndTimeStamp - a.epochEndTimeStamp,
-    );
+    const timeNow = Math.trunc(Date.now() / 1000);
 
-    let streak = 0;
-    for (let i = 0; i < sorted.length; i++) {
-      const current = sorted[i];
+    let isStreakBroken = false; // flag to break the streak
+    return earnedCheckpoints.reduce((streakCount, current, i) => {
+      if (isStreakBroken) return streakCount;
 
       // first iteration
       if (i === 0) {
-        const timeNow = Date.now() / 1000;
-
-        // multiplied by 2 to give a buffer of 2 days
-        const initialEpochGap = timeNow - current.epochEndTimeStamp;
+        const initialEpochGap = Math.trunc(timeNow - current.epochEndTimeStamp);
 
-        // if the last epoch was more than 1 day ago, break
-        if (initialEpochGap > ONE_DAY_IN_S) break;
+        // If the epoch gap is greater than the epoch length
+        if (initialEpochGap > Number(current.epochLength)) {
```

Comment:
can be a few seconds greater, depending on when backend decides to close the epoch, right? unsure if it'll be somehow noticed though